solr/contrib/jaegertracer-configurator/src/java/org/apache/solr/jaeger/JaegerTracerConfigurator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.jaeger;
+
+import io.jaegertracing.Configuration;
+import io.jaegertracing.samplers.ConstSampler;
+import io.opentracing.Tracer;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.TracerConfigurator;
+
+import static org.apache.solr.common.SolrException.ErrorCode.SERVER_ERROR;
+
+public class JaegerTracerConfigurator extends TracerConfigurator {
+  public static final String AGENT_HOST = "agentHost";
+  public static final String AGENT_PORT = "agentPort";
+  public static final String LOG_SPANS = "logSpans";
+  public static final String FLUSH_INTERVAL = "flushInterval";
+  public static final String MAX_QUEUE_SIZE = "maxQueueSize";
+
+  private volatile Tracer tracer;
+
+  @Override
+  public Tracer getTracer() {
+    return tracer;
+  }
+
+  @Override
+  public void init(NamedList args) {
+    Object host = args.get(AGENT_HOST);
+    if (!(host instanceof String)) {
+      throw new IllegalArgumentException("Expected a required string for param '" + AGENT_HOST + "'");
+    }
+
+    Object portArg = args.get(AGENT_PORT);
+    if (!(portArg instanceof Integer)) {
+      throw new IllegalArgumentException("Expected a required int for param '" + AGENT_PORT + "'");
+    }
+    int port = (Integer) portArg;
+
+    Boolean logSpans = args.getBooleanArg(LOG_SPANS);
+    if (logSpans == null)
+      logSpans = true;
+
+    Object flushIntervalArg = args.get(FLUSH_INTERVAL);
+    if (flushIntervalArg != null && !(flushIntervalArg instanceof Integer)) {
+      throw new IllegalArgumentException("Expected a required int for param '" + FLUSH_INTERVAL +"'");
+    }
+    int flushInterval = flushIntervalArg == null ? 5000 : (Integer) flushIntervalArg;
```

Comment:
Should the default flush interval be shorter? With the current defaults, Solr requests are limited to 2000 RPS before trace data will start being dropped. Of course, the person configuring Solr could set values instead of using the defaults for use cases requiring above 2000 RPS, but flushing every 1 second (or maybe even more frequently, like half a second) seems fairly safe as well.

Reply:
+1